### ISSUE TITLE
Borg Should Pursue Uniques More Aggressively

### ISSUE THREAD

**Issue by Netbrian** [angband/angband#6122]
I noticed that the borg will often refuse to dive because he wants to kill a particular unique first.  However, he won't actually try very hard to kill that unique.

For instance, in the below save, the borg wants to kill Harowen.  But if you try creating Harown in debug mode, he simply attacks once and teleports away.  The borg then loses interest and continues diving.  Even when he's at the dlvl limit, he'll sometimes say he got bored and leave the level.

My suggestion to start with would be to add some code that discourages the borg from leaving the dungeon level if he's trying to kill a unique he knows is around.

[Cirialf Harowen 01.zip](https://github.com/user-attachments/files/17576408/Cirialf.Harowen.01.zip)

@agoodman00

**Comment by agoodman00** [comment]
I agree but this gets "enhancement queued".  It will take a bunch of code to make sure that makes sure the borg sticks around but, at the same time, if it gets REALLY dangerous (say, he wants to kill a Unique that is way above his ability) he leaves and tries again when he is better prepared.  Or if the unique is hanging out with 12 AMHD or is in a vault with 3 other uniques, he needs to pick a better time.

**Comment by Netbrian** [comment]
Right, I understand this would be more iterative.  Maybe start with something like "don't powerdive or leave out of bordeom if the target unique is on the level"

The Druid borg seems to be really, really struggling with Harown specifically for some reason for some reason.

**Comment by Netbrian** [comment]
(Obviously I realize the complexity and that there are higher priorities, I was just thinking aloud.)

**Comment by agoodman00** [comment]
I think https://github.com/angband/angband/pull/6584 may help this a bunch.  It was not recognizing Uniques when determining danger in code that was designed to often make the borg a bit more aggressive around them.

**Comment by Netbrian** [comment]
OK, I'll give it a shot!  I haven't been running the borg as much because I've been playing Angband myself more often recently, but I'll see what happens -- especially in "cheat death" mode.
